src/main/java/domain/Cars.java

```diff
@@ -0,0 +1,39 @@
+package domain;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class Cars {
+    private final List<Car> cars;
+
+    public Cars(List<Car> cars) {
+        validateCarAmount(cars);
+        validateDuplicateName(cars);
+        this.cars = cars;
+    }
+
+    private void validateCarAmount(List<Car> cars) {
+        if (cars.size() == 1) {
+            throw new IllegalArgumentException(ErrorMessage.MULTIPLE_CARS_REQUIRED.getMessage());
+        }
+    }
+
+    private void validateDuplicateName(List<Car> cars) {
+        int duplication = 0;
+        Set<String> validateCar = new HashSet<>();
+        for (Car car : cars) {
+            validateCar.add(car.getCarName());
+            duplication = validateCar.size();
+        }
+
+        if (cars.size() != duplication) {
+            throw new IllegalArgumentException(ErrorMessage.DUPLICATE_NAME.getMessage());
+        }
```

Comment:
```suggestion
        Set<Car> validateCar = new HashSet<>(cars);

        if (validateCar.size() != cars.size()) {
            throw new IllegalArgumentException(ErrorMessage.DUPLICATE_NAME.getMessage());
        }
```
equals and hashcode 에 대해서 학습해보면 좋을것 같아요!

Reply:
<equals and hashcode 에 대해서 학습한 것>
1. List<Car> cars 를 Set<Car> 로 넣는 과정에서 중복 검사할 때, equals and hashcode 를 통한다.
2. equals and hashcode 는 객체의 고유한 주소 값을 리턴하여 비교를 한다.
3. new Car(String name) 을 통해 생성된 객체는 name 이 같더라도 주소 값이 다르므로 다른 객체다.
4. 따라서 Car 클래스에 equals and hashcode 를 재정의하여 사용해야 한다.